x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/logging/CppLogMessageHandler.java

```diff
@@ -283,6 +283,15 @@ private void parseMessage(XContent xContent, BytesReference bytesRef) {
             if (upstreamMessage.contains("bad_alloc")) {
                 upstreamMessage += ", process ran out of memory.";
             }
+
+            // add version information, so it's conveniently next to the crash log
+            upstreamMessage += ", version: ";
+            try {
+                upstreamMessage += getCppCopyright(Duration.ofMillis(10));
```

Comment:
It might look a bit weird to see the copyright as part of the "version" part of the message.  The bit we actually want comes before the copyright.  It could be got rid of quite simply like this:

```
    upstreamMessage += getCppCopyright(Duration.ofMillis(10)).replaceFirst(" Copyright.*", "");
```

For example, from:

```
autodetect (64 bit): Version 7.0.0-alpha1-SNAPSHOT (Build 799d6b9e8bdff9) Copyright (c) 2018 Elasticsearch BV
```

we'd add:

```
, version: autodetect (64 bit): Version 7.0.0-alpha1-SNAPSHOT (Build 799d6b9e8bdff9)
```

If we can already easily distinguish `normalize` from `autodetect` then we could add another `replaceFirst` to strip up to "Version ", so we'd get:

```
, version: 7.0.0-alpha1-SNAPSHOT (Build 799d6b9e8bdff9)
```

But it's useful to know the program name and I can't remember how easy it is to find that out from what's in the error message already.

Reply:
The process name is part of the log line, e.g.

```
[2018-05-17T10:31:56,396][ERROR][o.e.x.m.j.p.a.NativeAutodetectProcess] [g6] autodetect process stopped unexpectedly: Fatal error: 'si_signo 11, si_code: 1, si_errno: 0, address: 0x7f5b8d4160b8, library: /home/hendrik/work/git-elastic/elasticsearch/test-install/elasticsearch-7.0.0-alpha1-SNAPSHOT/modules/x-pack/x-pack-ml/platform/linux-x86_64/bin/../lib/libMlMaths.so, base: 0x7f5b8d0de000, normalized address: 0x3380b8', version: autodetect (64 bit): Version based on 7.0.0-alpha1-SNAPSHOT (Build DEVELOPMENT BUILD by hendrik) Copyright (c) 2018 Elasticsearch BV
```

therfore: yes we can potentially replace it.

Reply:
It might be useful then to have a specific `getCppVersion` method that pulls out just the version data as @droberts195 describes

Reply:
It might also be a good opportunity to update the comment on the `getCppCopyright` method as it currently refers to  "... the process ID of the C++ process ".

Reply:
> It might be useful then to have a specific getCppVersion method that pulls out just the version data

If we do that it would be good to use it instead of `getCppCopyright()` in `NativeController.getNativeCodeInfo()`.

Reply:
:+1: I am refactoring it, so there is just 1 place that parses the string.